util/src/hash.rs

```diff
@@ -548,51 +527,44 @@ impl<'a> From<&'a H256> for U256 {
 
 impl From<H256> for Address {
 	fn from(value: H256) -> Address {
-		unsafe {
-			let mut ret: Address = ::std::mem::uninitialized();
-			::std::ptr::copy(value.as_ptr().offset(12), ret.as_mut_ptr(), 20);
-			ret
-		}
+		let mut ret = Address::new();
+		ret.0.copy_from_slice(&value[12..32]);
+		ret
 	}
 }
 
 impl From<H256> for H64 {
 	fn from(value: H256) -> H64 {
-		unsafe {
-			let mut ret: H64 = ::std::mem::uninitialized();
-			::std::ptr::copy(value.as_ptr().offset(20), ret.as_mut_ptr(), 8);
-			ret
-		}
+		let mut ret = H64::new();
+		ret.0.copy_from_slice(&value[20..28]);
+		ret
 	}
 }
+
 /*
 impl<'a> From<&'a H256> for Address {
 	fn from(value: &'a H256) -> Address {
-		unsafe {
-			let mut ret: Address = ::std::mem::uninitialized();
-			::std::ptr::copy(value.as_ptr().offset(12), ret.as_mut_ptr(), 20);
+			let mut ret = Address::new();
+			ret.0.copy_from_slice(&value[12..32]);
```

Comment:
Hm, it can't hurt much to have both impls -- I would opt to keep it as long as everything compiles with it.